### PR TITLE
feat(starters): Update default starter

### PR DESCRIPTION
**Pull request by fk**
Preview: https://starterdefaultv2.gatsbyjs.io/

## Description

This PR updates `gatsby-starter-default` to reflect the look-and-feel of the `minimal` starter (used for `create-gatsby`). It also adds some hopefully helpful "next steps" starting points to the index page of the starter.

- drop a good chunk of what looks like a version of `normalize.css` styles
- add CSS vars – helps with consistency esp. WRT `style`
- add two sets of links to Tutorial, Documentation, etc.
- various content/copy changes
- just like the `minimal` starter it uses inline `style`s for the index page – wondered about using CSS modules instead
- moves a couple things to the "global" `layout.css`, most significantly link and `code` styles

### CODE REVIEW

File: starters/default/src/pages/using-typescript.tsx [hunk 1]
```diff
@@ -14,18 +14,21 @@ type DataProps = {
...
-    <h1>Gatsby supports TypeScript by default!</h1>
+    <h1>
+      Gatsby supports <b>TypeScript by default</b>
+    </h1>
     <p>
-      This means that you can create and write <em>.ts/.tsx</em> files for your
-      pages, components etc. Please note that the <em>gatsby-*.js</em> files
-      (like gatsby-node.js) currently don't support TypeScript yet.
+      This means that you can create and write <code>.ts/.tsx</code> files for
+      your pages, components etc. Please note that the <code>gatsby-*.js</code>{" "}
```
Comment (LekoArts) [inline starters/default/src/pages/using-typescript.tsx:22]: We can remove this note about gatsby-*.js files now 🎉

Comment (fk) [reply starters/default/src/pages/using-typescript.tsx]: Thank you, Lennart! 🤗
I tweaked the copy, LMK what you think – too much?

File: starters/default/src/pages/using-typescript.tsx [hunk 2]
```diff
@@ -14,18 +14,20 @@ type DataProps = {
...
     </p>
     <p>
-      For type checking you'll want to install <em>typescript</em> via npm and
-      run <em>tsc --init</em> to create a <em>tsconfig</em> file.
+      For type checking you'll want to install <code>typescript</code> via npm
+      and run <code>tsc --init</code> to create a <code>tsconfig</code> file.
     </p>
     <p>
-      You're currently on the page "{path}" which was built on{" "}
+      You're currently on the page <code>{path}</code> which was built on{" "}
```
Comment (wardpeet) [inline starters/default/src/pages/using-typescript.tsx:30]:
```suggestion
      You're currently on the page <code>{location.pathname}</code> which was built on{" "}
```

File: starters/default/src/pages/index.js [hunk 3]
```diff
@@ -5,26 +5,148 @@ import { StaticImage } from "gatsby-plugin-image"
...
-    </p>
+    <div style={{ textAlign: "center" }}>
+      <StaticImage
+        src="../images/example.png"
+        width={64}
+        quality={95}
+        formats={["auto", "webp", "avif"]}
+        alt=""
+        style={{ marginBottom: `var(--space-3)` }}
+      />
```
Comment (wardpeet) [inline starters/default/src/pages/index.js:113]:
```suggestion
      <StaticImage
        src="../images/example.png"
        loading="eager"
        width={64}
        quality={95}
        formats={["auto", "webp", "avif"]}
        alt=""
        style={{ marginBottom: `var(--space-3)` }}
      />
```

**Review by wardpeet — COMMENTED** [review]
smalll nits

File: starters/default/src/pages/using-typescript.tsx [hunk 4]
```diff
@@ -14,18 +14,20 @@ type DataProps = {
 const UsingTypescript: React.FC<PageProps<DataProps>> = ({ data, path }) => (
```
[the remark this answers is not in the document]
Comment (fk) [reply starters/default/src/pages/using-typescript.tsx:14]: Let's try!

**Comment by fk** [comment]
Any opinions about CSS modules vs the current inline `style`?

**Comment by wardpeet** [comment]
No real opinion on it. If you use css-modules, your middot piece can be moved to css

**Comment by fk** [comment]
I just followed what the minimal starter does - but we have CSS modules in the Tutorial, no extra dependencies, it's more approachable ...

**Comment by pragmaticpat** [comment]
@wardpeet and @fk - is there anything else needed for us to get this merged in?

Thanks for the refinements so far!

**Pull request merged**